packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -125,32 +59,35 @@ function isAction<A extends AnyAction>(
   return action.type === type;
 }
 
-export type QueryBarFormField = {
-  value: unknown;
-  string: string;
-  valid: boolean;
-};
-
 export type QueryBarState = {
-  fields: Record<QueryProperty, QueryBarFormField>;
+  fields: QueryFormFields;
   expanded: boolean;
   serverVersion: string;
   schemaFields: unknown[];
-  lastAppliedQuery: unknown | null;
+  lastAppliedQuery: BaseQuery | null;
   /**
    * For testing purposes only, allows to track whether or not apply button was
    * clicked or not
    */
   applyId: number;
+  activeTab: QueryHistoryTab;
+  namespace: string;
+  host?: string;
+  recentQueries: RecentQuery[];
+  favoriteQueries: FavoriteQuery[];
 };
 
 export const INITIAL_STATE: QueryBarState = {
-  fields: mapQueryToValidQueryFields(DEFAULT_FIELD_VALUES),
+  fields: mapQueryToFormFields(DEFAULT_FIELD_VALUES),
   expanded: false,
   serverVersion: '3.6.0',
```

Comment:
Now that we are [officially at 4.2](https://www.mongodb.com/docs/compass/master/install/#software-requirements), should we bump this up across?